### PR TITLE
Remove UnknownEncodingMessageReader and replace with exceptions #262

### PR DESCRIPTION
**Pull request by smadasu**
Removed UNKNOWN encoding and replaced class with Exception. Please approve.

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
Can you fix the dco? If you look in the details, it shows you how

File: core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java [hunk 1]
```diff
@@ -57,7 +59,7 @@ public static MessageReader parseStructuredOrBinaryMessage(
             return binaryMessageFactory.apply(SpecVersion.parse(specVersionUnparsed));
         }
 
-        return unknownMessageFactory.get();
+        throw CloudEventRWException.newOther(new IllegalStateException("Invalid content type or spec version"));
```
Comment (slinkydeveloper) [inline core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java:62]: There is a CloudEventRWException kind for it right? If not, can we create it?

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]
Can you add a test for `parseStructuredOrBinaryMessage` that tests all three cases (encoding binary, encoding structured and exception thrown for unknown encoding)?

File: core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java (hunk 1 above)
Comment (smadasu) [reply core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java:62]: I will fix the dco. As for that exception, this call is creating that exception

Comment (smadasu) [reply core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java:62]: I will work on the test

Comment (slinkydeveloper) [reply core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java:62]: I would love to have a specific `CloudEventRWException` for this, adding a kind here: https://github.com/cloudevents/sdk-java/blob/master/api/src/main/java/io/cloudevents/rw/CloudEventRWException.java#L28

File: api/src/main/java/io/cloudevents/rw/CloudEventRWException.java [hunk 2]
```diff
@@ -56,6 +56,10 @@
          * Error while converting CloudEventData.
          */
         DATA_CONVERSION,
+        /**
+         * Invalid content type or spec version
+         */
+        PARSE_EXCEPTION,
```
Comment (slinkydeveloper) [inline api/src/main/java/io/cloudevents/rw/CloudEventRWException.java:62]: I think this should be named `UNKNOWN_ENCODING`, `PARSE_EXCEPTION` is too generic.

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/cloudevents/rw/CloudEventRWException.java (hunk 2 above)
Comment (smadasu) [reply api/src/main/java/io/cloudevents/rw/CloudEventRWException.java:62]: OK. Sure

Comment (smadasu) [reply api/src/main/java/io/cloudevents/rw/CloudEventRWException.java]: I have updated everything requested but the build is still failing for some reason. Please look into it.

**Comment by slinkydeveloper** [comment]
SGTM, can you fix the broken tests?

**Comment by slinkydeveloper** [comment]
Maybe you need a rebase on latest upstream/master

**Comment by slinkydeveloper** [comment]
Thanks for the contribution, I solved the broken test!

**Pull request merged**